controller/playground/manager.py

```diff
@@ -0,0 +1,349 @@
+
+EVALUATION_RUN_LIMIT_DEFAULT = 100
+
+
+class EVALUATION_RUN_STATE:
+    INITIATED = "INITIATED"
+    RUNNING = "RUNNING"
+    SUCCESS = "SUCCESS"
+    FAILED = "FAILED"
+
+
+def get_search_result_for_text(
+    project_id: str,
+    user_id: str,
+    embedding_id: str,
+    question: str,
+    limit: int,
+    filter=None,
+    threshold=None,
+    save_question=None,
+):
+    question_tensor = __get_tensors_for_texts(project_id, embedding_id, [question])
+    if not question_tensor or not question_tensor[0]:
+        return []
+
+    records = __get_most_similar_records(
+        project_id,
+        embedding_id,
+        question_tensor[0],
+        limit,
+        filter,
+        threshold,
+        save_question,
+        question,
+        user_id,
+    )
+
+    record_ids = [record["id"] for record in records]
+    record_obj_map = record_db_bo.get_full_record_data_for_id_group(
+        project_id, record_ids
+    )
+
+    return [
+        {
+            "data": record_obj_map.get(record["id"]),
+            "id": record["id"],
+            "score": record["score"],
+        }
+        for record in records
+    ]
+
+
+def create_evaluation_set(
+    project_id: str, question: str, record_ids: List[str], created_by: str
+):
+    evaluation_set_db_bo.create(project_id, question, created_by, record_ids, True)
+
+
+def delete_evaluation_sets(project_id: str, set_ids: str):
+    evaluation_set_db_bo.delete_all(project_id, set_ids, True)
+
+
+def get_evaluation_set_by_id(project_id: str, set_id: str):
+    return evaluation_set_db_bo.get(project_id, set_id)
+
+
+def get_evaluation_sets(project_id: str):
+    return evaluation_set_db_bo.get_all(project_id)
+
+
+def get_evaluation_sets_by_group_id(project_id: str, evaluation_group_id: str):
+    return evaluation_set_db_bo.get_by_evaluation_group_id(
+        project_id, evaluation_group_id
+    )
+
+
+def create_evaluation_group(
+    project_id: str, name: str, evaluation_set_ids: List[str], created_by: str
+):
+    evaluation_group_db_bo.create(
+        project_id, name, created_by, evaluation_set_ids, True
+    )
+
+
+def delete_evaluation_groups(project_id: str, group_ids: str):
+    evaluation_group_db_bo.delete_all(project_id, group_ids, True)
+
+
+def get_evaluation_group_by_id(project_id: str, group_id: str):
+    return evaluation_group_db_bo.get(project_id, group_id)
+
+
+def get_evaluation_groups(project_id: str):
+    return evaluation_group_db_bo.get_all(project_id)
+
+
+def delete_evaluation_runs(project_id: str, run_ids: str):
+    return evaluation_run_db_bo.delete_all(project_id, run_ids, True)
+
+
+def get_evaluation_runs(project_id: str):
+    evaluation_runs_objects = evaluation_run_db_bo.get_all(project_id)
+    evaluation_runs = []
+    for evaluation_run in evaluation_runs_objects:
+        results = to_frontend_obj_raw(evaluation_run.results)
+        evaluation_runs.append(
+            {**sql_alchemy_to_dict(evaluation_run, True), "results": results}
+        )
```

Comment:
dublicated code (get result to dict) maybe as small function + list comprehension